sdks/unreal/Agones/Source/Agones/Classes/Classes.h

```diff
@@ -0,0 +1,361 @@
+
+	FHealth()
+	{
+	}
+
+	explicit FHealth(const TSharedPtr<FJsonObject> JsonObject)
+	{
+		JsonObject->TryGetBoolField(TEXT("disabled"), bDisabled);
+		JsonObject->TryGetNumberField(TEXT("period_seconds"), PeriodSeconds);
+		JsonObject->TryGetNumberField(TEXT("failure_threshold"), FailureThreshold);
+		JsonObject->TryGetNumberField(TEXT("initial_delay_seconds"), InitialDelaySeconds);
+	}
+};
+
+USTRUCT(BlueprintType)
+struct FSpec
+{
+	GENERATED_BODY()
+
+	UPROPERTY(BlueprintReadOnly)
+	FHealth Health;
+
+	FSpec()
+	{
+	}
+
+	explicit FSpec(const TSharedPtr<FJsonObject> JsonObject)
+	{
+		const TSharedPtr<FJsonObject>* HealthJsonObject;
+		if (JsonObject->TryGetObjectField(TEXT("health"), HealthJsonObject))
+		{
+			Health = FHealth(*HealthJsonObject);
+		}
+	}
+};
+
+USTRUCT(BlueprintType)
+struct FPort
+{
+	GENERATED_BODY()
+
+	UPROPERTY(BlueprintReadOnly)
+	FString Name;
+
+	UPROPERTY(BlueprintReadOnly)
+	int32 Port = 0;
+
+	FPort()
+	{
+	}
+
+	explicit FPort(const TSharedPtr<FJsonObject> JsonObject)
+	{
+		JsonObject->TryGetStringField(TEXT("name"), Name);
+		JsonObject->TryGetNumberField(TEXT("port"), Port);
+	}
+};
+
+USTRUCT(BlueprintType)
+struct FStatus
+{
+	GENERATED_BODY()
+
+	UPROPERTY(BlueprintReadOnly)
+	FString State;
+
+	UPROPERTY(BlueprintReadOnly)
+	FString Address;
+
+	UPROPERTY(BlueprintReadOnly)
+	TArray<FPort> Ports;
+
+	FStatus()
+	{
+	}
+
+	explicit FStatus(const TSharedPtr<FJsonObject> JsonObject)
+	{
+		JsonObject->TryGetStringField(TEXT("state"), State);
+		JsonObject->TryGetStringField(TEXT("address"), Address);
+		const TArray<TSharedPtr<FJsonValue>>* PortsArray;
+		if (JsonObject->TryGetArrayField(TEXT("ports"), PortsArray))
+		{
+			const int32 ArrLen = PortsArray->Num();
+			for (int32 i = 0; i < ArrLen; ++i)
+			{
+				const TSharedPtr<FJsonValue>& PortItem = (*PortsArray)[i];
+				if (PortItem.IsValid() && !PortItem->IsNull())
+				{
+					FPort Port = FPort(PortItem->AsObject());
+					Ports.Add(Port);
+				}
+			}
+		}
+	}
+};
+
+USTRUCT(BlueprintType)
+struct FGameServerResponse
+{
+	GENERATED_BODY()
+
+	UPROPERTY(BlueprintReadOnly)
+	FStatus Status;
+
+	UPROPERTY(BlueprintReadOnly)
+	FObjectMeta ObjectMeta;
+
+	UPROPERTY(BlueprintReadOnly)
+	FSpec Spec;
+
+	FGameServerResponse()
+	{
+	}
+
+	explicit FGameServerResponse(const TSharedPtr<FJsonObject> JsonObject)
+	{
+		const TSharedPtr<FJsonObject>* ObjectMetaJsonObject;
+		if (JsonObject->TryGetObjectField(TEXT("object_meta"), ObjectMetaJsonObject))
+		{
+			ObjectMeta = FObjectMeta(*ObjectMetaJsonObject);
+		}
+		const TSharedPtr<FJsonObject>* SpecJsonObject;
+		if (JsonObject->TryGetObjectField(TEXT("spec"), SpecJsonObject))
+		{
+			Spec = FSpec(*SpecJsonObject);
+		}
+		const TSharedPtr<FJsonObject>* StatusJsonObject;
+		if (JsonObject->TryGetObjectField(TEXT("status"), StatusJsonObject))
+		{
+			Status = FStatus(*StatusJsonObject);
+		}
+	}
+};
+
+USTRUCT(BlueprintType)
+struct FKeyValuePair
+{
+	GENERATED_BODY()
+
+	UPROPERTY()

```

Comment:
Should this also be `UPROPERTY(BlueprintReadOnly)`?

Reply:
Yep, just did a check and I only set it in C++ it is passed in as a argument in the func call, will check the rest of the file as well.